aziotctl/src/internal/check/checks/read_key_pairs.rs

```diff
@@ -82,23 +85,54 @@ impl ReadKeyPairs {
                 })?;
                 Ok::<_, anyhow::Error>(key)
             };
+
             match key() {
                 Ok(key) => {
+                    let key_length = key.bits();
+
+                    match key.id() {
```

Comment:
Since you need `ec_key` per my other comment, it's easier to do `if let Ok(rsa) = key.rsa() { ... } else if let Ok(ec_key) = key.ec_key() { ... } else { Err(...) }` which combines the type check with getting the type-specific parameters.

With `.rsa()` you get the modulus length using `rsa.size()`, but note that it's in bytes, not bits.

Reply:
Fixed